### DISCUSSION TITLE
lamella identifiers

### DISCUSSION THREAD

**Discussion by xshang93** [neperfepx/neper#492] [Q&A] [answered]
Hi @rquey , in the .tess file, the `*lam` "denotes the beginning of an optional lamella identifier field". If the lam_width is define as `width=A:B`, does this mean that 1 always denotes a lamella with width A and 2 denotes B, regardless which one is at the starting position in a cell?

I'm using `-group lam==1?1:2` to group HCP and BCC crystals for simulations in FEPX and I wanted to confirm that this is the correct way to do it. I'm using` pos=random` in the lam settings.

Thank you!

(post edited for better clarity)

**Comment by rquey** [comment, marked as answer]
Yes, your understanding is correct. You could even use `-group lam`...

**Reply by xshang93** [reply]
Thank you. A follow up question: when assigning orientations, how to make sure that the orientations for different phases are correctly assigned, if the starting position is random? I'm currently assigning the orientations with a file for each cell, within the file there are two alternating orientations:

293.204447 125.040526 87.767133 
21.462792 87.518335 260.078253 
293.204447 125.040526 87.767133 
21.462792 87.518335 260.078253 
293.204447 125.040526 87.767133 
21.462792 87.518335 260.078253

How do I make sure that, for example, `293.204447 125.040526 87.767133 ` is assigned to group 2 while the other one is assigned to group 1, since we don't know which one is at the starting position before running the tessellation?

**Reply by rquey** [reply]
I would proceed in two steps, the first one to compute the morphology and the second one to do the same and set the orientations. Here is how on a small tessellation (and assuming you want to use Euler angles):

```
neper -T -n 2::from_morpho -morpho "voronoi::lamellar(w=0.1:0.2)" -statcell "scaleid(1),lam" -o d

awk '{if ($1==1) print $2}' d.stcell > ori-grain1
awk '{if ($1==2) print $2}' d.stcell > ori-grain2
sed -i "s/^\<1\>/104.5 15.2 32.9/" ori-grain1
sed -i "s/^\<2\>/212.6 57.0 58.1/" ori-grain1
sed -i "s/^\<1\>/ 43.9 42.0 18.2/" ori-grain2
sed -i "s/^\<2\>/147.8 37.8 83.6/" ori-grain2

rm -f ori
echo "1 file(ori-grain1,des=euler-bunge)" >> ori
echo "2 file(ori-grain2,des=euler-bunge)" >> ori

neper -T -n 2::from_morpho -morpho "voronoi::lamellar(w=0.1:0.2)" -ori "random::msfile(ori)" -o d
```

Visualize:
```
neper -V d.tess -datacellcol ori -print img
```
![image](https://user-images.githubusercontent.com/2659376/172920239-1d2eadc9-dff8-4d8c-9102-b498e1792e97.png)

**Reply by xshang93** [reply]
Thanks. I just had time to test it today and this works great.
